youtube_dl/extractor/youtube.py

```diff
@@ -1825,7 +1825,7 @@ def extract_player_response(player_response, video_id):
         # Get video info
         video_info = {}
         embed_webpage = None
-        if re.search(r'player-age-gate-content">', video_webpage) is not None:
+        if self._html_search_meta('og:restrictions:age', video_webpage) == "18+":
```

Comment:
Since your fix uses `_html_search_meta`, non-fatal warnings are sometimes now issued for non-age gated files.

I changed it to